internal/handlers/common/filter_iterator.go

```diff
@@ -52,7 +59,21 @@ func (iter *filterIterator) Next() (struct{}, *types.Document, error) {
 			return unused, nil, lazyerrors.Error(err)
 		}
 
-		matches, err := FilterDocument(doc, iter.filter)
+		var matches bool
+		if iter.filter.Has("$expr") {
```

Comment:
I think this logic is more suited in `filter.go`.

Reply:
This sounds like something to discuss. I like the current structure but I'm not sure I fully understand all the use cases.